libs/labelbox/src/labelbox/schema/dataset.py

```diff
@@ -912,3 +913,80 @@ def _convert_items_to_upsert_format(self, _items):
             }  # remove None values
             _upsert_items.append(DataRowUpsertItem(payload=item, id=key))
         return _upsert_items
+    
+    def set_iam_integration(self, iam_integration: Union[str, IAMIntegration, None] = None) -> Optional[IAMIntegration]:
+          """
+          Sets the IAM integration for the dataset. IAM integration is used to sign URLs for data row assets.
+
+          >>>    # Get all IAM integrations
+          >>>    iam_integrations = client.get_organization().get_iam_integrations()
+          >>>    
+          >>>    # Get IAM integration id
+          >>>    iam_integration_id = [integration.uid for integration
+          >>>      in iam_integrations
+          >>>      if integration.name == "My S3 integration"][0]
+          >>>
+          >>>   # Set IAM integration for integration id
+          >>>   dataset.set_iam_integration(iam_integration_id)
+          >>>
+          >>>    # Get IAM integration object
+          >>>    iam_integration = [integration.uid for integration
+          >>>      in iam_integrations
+          >>>      if integration.name == "My S3 integration"][0]
+          >>>
+          >>>   # Set IAM integration for IAMIntegrtion object
+          >>>   dataset.set_iam_integration(iam_integration)
+          >>>
+          >>>   # Unset IAM integration
+          >>>   dataset.set_iam_integration()
+          """
+
+          # Unset IAM integration if iam_integration is None
+          if iam_integration is None:
```

Comment:
I'd rather this method was split into two, `set_iam_integration` and `remove_iam_integration`, and the iam_integration argument was made required.

Reply:
Actually, could you do `add_iam_integration` and `remove_iam_integration`? We don't have a standard for this yet, but I'd like to start having 'create/delete' semantics for creating/deleting objects, and `add/remove` for creating/deleting associations between objects.

Reply:
Done